lib/bap/bap.mli

```diff
@@ -3822,6 +3860,377 @@ module Std : sig
+    val split_bot : t -> t * t
+
+    (** [elts ~rev blk] return all elements of the [blk].
+        if [rev] is false or left unspecified, then elements are returned
+        in the following order: $\Phi$-nodes, defs (in normal order),
+        jmps in the order in which they will be taken.
+        If [rev] is true, the order will be the following:
+        all jumps in the opposite order, then definitions in the
+        opposite order, and finally $\Phi$-nodes. *)
+    val elts : ?rev:bool -> t -> elt seq
+
+    (** [dominated blk by:dom def] if [def] is dominated by [dom] in [blk].  *)
+    val dominated : t -> by:tid -> tid -> bool
+
+    (** Builder interface.  *)
+    module Builder : sig
+      (** This interface provides an efficient way to build new
+          blocks. It is also useful, when rebuilding existing block,
+          as it allows to specify the [tid] of the block. It is a user
+          responsibility to preserve the uniqueness of tidentifiers
+          throughout the program instance.
+      *)
+
+      type t
+
+      (** [create ~tid ~phis ~defs ~jmp ()] creates a block builder.
+          If [tid] parameter is specified, then the new block will have
+          this tid. If any of [phis], [defs] or [jmps] parameters are
+          specified, the provtided number would be used as a hint of the
+          expected amount of the corresponding entries. Since it is the
+          hint, it can mismatch with the actual size. The hint must be
+          a positive number.
+      *)
+      val create : ?tid:tid -> ?phis:int -> ?defs:int -> ?jmps:int -> unit -> t
+      val add_def : t -> def term -> unit
+      val add_jmp : t -> jmp term -> unit
+      val add_phi : t -> phi term -> unit
+      val add_elt : t -> elt -> unit
+      val result  : t -> blk term
+    end
+
+    include Regular with type t := t
+  end
+
+  module Def : sig
+    type t = def term
+    val create : var -> exp -> t
+    val lhs : t -> var
+    val rhs : t -> exp
+    val with_lhs : t -> var -> t
+    val with_rhs : t -> exp -> t
+
+    include Regular with type t := t
+  end
+
+
+  module Jmp : sig
+    type t = jmp term
+    (** [create ?cond ?return target]  *)
+    val create_call : ?cond:exp -> call -> t
+    val create_goto : ?cond:exp -> label -> t
+    val kind : t -> jmp_kind
+    val cond : t -> exp
+
+    val with_cond : t -> exp -> t
+    val with_kind : t -> jmp_kind -> t
+
+    include Regular with type t := t
+  end
+
+  module Phi : sig
+    type t = phi term
+    val create : var -> def term -> t
+    val lhs : t -> var
+    val defs : t -> def term seq
+    val add_def : t -> def term -> t
+    val remove_def : t -> tid -> t
+
+    include Regular with type t := t
+  end
+
+  module Arg : sig
+    type t = arg term
+
+    val create : ?name:string -> typ -> t
+    val of_var : var -> t
+    val to_var : t -> var
+    val name : t -> string
+    include Regular with type t := t
+  end
+
+  module Call : sig
+    type t = call
```

Comment:
I never heard anything about inter-procedural jumps with their own stack, google also didn't help me. 
We have `goto` for local jumps, and `call` for leaving the procedure. Without explicit return edge, it will not be possible to write intra-procedural analysis.